app/assets/javascripts/editor.js

```diff
@@ -85,45 +85,50 @@ function snippet_changed(item){
   $('#association').toggle(!checked);
 }
 
-function create_editor(item) {
-  item.parent().prepend("<div id='editor1'></div>");
-  item.hide();
-  Editor = ace.edit("editor1");
+function create_editor() {
+  var editorId = "editor" + Math.floor((Math.random() * 100) + 1),
```

Comment:
Where is there the chance of collisions?  As far as I can tell, it would only be on the class edit form, which as per the other comments, is a bad idea anyway.